prime-router/docs/authz/authz-design.md

```diff
@@ -168,13 +168,57 @@ fun setApplicationProfile(
 }
 ```
 
-We could go even further and write various APIs (and potentially CLI calls) that will set up an application user from 
-scratch. This would dramatically simplify the process of setting up a sender since we would not need to handle going
-back and forth between API and Okta Admin UI and could provide the necessary values needed for a new sender.
-We could also reuse our existing authentication and authorization within the project to ensure only specific people are able to 
-hit these endpoints.
-
-Sample endpoints:
+We have several viable approaches to handling updates to senders:
+* Build a process within the auth project to retrieve an application user's group memberships and update their profile
+  accordingly. 
+  * This allows group management to occur within the Okta Admin Console, but a trigger to perform this step is needed.
+    * We can build a API call (triggerable via frontend and/or CLI) that takes a client ID of a sender as input and does
+      the needed profile updates on demand.
+    * We can configure a schedule trigger to scan all application users and update their profiles based on their Okta
+      group memberships periodically. This removes the potential for group memberships and sender profiles to fall out
+      of sync, but the refresh will happen on a delay and this would require the microservice to perform more work on an
+      ongoing basis.
+    * Using Okta event hooks to send a POST request to the auth microservice whenever an application's group membership
```

Comment:
Not really relevant to the review itself, but (from my less-informed opinion) this is the option that makes the most sense to me. It sounds like once it is running it would have the least amount of administrative hassle when onboarding.